### PR TITLE
feature: added the duel ui

### PR DESCRIPTION
**Pull request by YoMoM829**
This pull request contains the UI for a Duel. This shows the buildup of a Duel, so the Rock, Paper and Shazam stages. This also shows the results of the Duel including the move both players made. Additionally, this UI is responsive to a mobile and desktop screen.

### CODE REVIEW

**Comment by RoryT-U** [comment]
Please make sure your next branch uses the correct format and is slightly more descriptive (gameUI2 is that specific, something like feature/duel-ui would be good).

File: client/player.ts
Comment (RoryT-U) [inline client/player.ts:1]: The client should not have the entire Player class, I am assuming this is for testing purpose? Which if it is, make sure to remove if before you start a Pull Request.

File: client/src/App.tsx
Comment (RoryT-U) [inline client/src/App.tsx:1]: I am assuming these changes were also for testing? If so please remove them back before sending PR.

**Comment by RoryT-U** [comment]
There may be some issues either on my side or in the code with the how the UI is responsive. If a few other people can test it and see how if it works that would be good.
image

File: client/src/components/DuelName.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        </div>
+      </div>
+      <div>
+        <div className="fixed bottom-5 left-8 text-2xl text-white font-bold font-sans bg-lime-600 rounded-md px-7 py-3">
+          {name2}
+          <div className="flex flex-row justify-between gap-x-4">
+            <img style={{height: 30}} className="object-fit" src={star}/>
+            <img style={{height: 30}} className="object-fit" src={star}/>
+            <img style={{height: 30}} className="object-fit" src={star}/>
+          </div>
```
Comment (RoryT-U) [inline client/src/components/DuelName.tsx:38]: Could potentially make these into components. Atleast atleast the inner part could be.

File: client/src/components/DuelScore.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <p className="text-8xl text-white font-bold font-sans">{matchPhrase}</p>
+      <div className="flex justify-center">
+        <p className="text-6xl text-white font-bold font-sans">{player2Score} - <span className="text-green-600">{player1Score}</span></p>
+      </div>
+    </div>
+  }
+  
+}
+
+const duelScore = ({player1Score, player2Score, matchResult}: PlayerScoreProps) => {
```
Comment (RoryT-U) [inline client/src/components/DuelScore.tsx:35]: Should be CamelCase (DuelScore)

File: client/src/components/DuelMove.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import rockLose from "../assets/Duel/Rock-Outline.svg"
+import scissorsWin from "../assets/Duel/Scissors.svg"
+import scissorsLose from "../assets/Duel/Scissors-Outline.svg"
+
+interface MoveProps {
+  type: string;
+  win: boolean;
+}
+
+const duelMove = ({type, win}: MoveProps) => {
```
Comment (RoryT-U) [inline client/src/components/DuelMove.tsx:16]: Should be CamelCase

File: client/src/components/DuelLoad.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+import shazamLogo from "../assets/Duel/SHAZAM.svg"
+import closed from "../assets/Duel/Hand.svg";
+import explosion from "../assets/Duel/Explosion.svg";
+
+interface LoadProps {
+  phase: number;
+}
+
+const duelLoad = ({phase}: LoadProps) => {
```
Comment (RoryT-U) [inline client/src/components/DuelLoad.tsx:12]: Should be CamelCase

**Review by RoryT-U — CHANGES_REQUESTED** [review]
Please have a look over the comments and make some appropriate changes.

**Comment by lkog77** [comment]
> There may be some issues either on my side or in the code with the how the UI is responsive. If a few other people can test it and see how if it works that would be good. image

Same

**Comment by RoryT-U** [comment]
Please resolve issues that you fix, or add a comment if you disagree with what I said.

**Comment by YoMoM829** [comment]
Fixed previous errors. UI should be responsive to when a screen is minimised on a desktop and should be able to work on different screen sizes.

**Comment by YoMoM829** [comment]
I reviewed the changes and added the following corrections. Changes were much appreciated. I haven't added types yet since I want to see if the UIs we made can be combined. If so, we should incorporate them and then types. Also, adding types should not take long at all.

File: client/player.ts
Comment (YoMoM829) [reply client/player.ts:1]: Yeah this is for testing sorry.

File: client/src/App.tsx
Comment (YoMoM829) [reply client/src/App.tsx:1]: I've changed it back so it should be fine.

File: client/src/components/DuelMove.tsx (hunk 3 above)
Comment (YoMoM829) [reply client/src/components/DuelMove.tsx]: Yup done that

File: client/src/components/DuelLoad.tsx (hunk 4 above)
Comment (YoMoM829) [reply client/src/components/DuelLoad.tsx]: Yup changed that.

File: client/player.ts
Comment (RoryT-U) [reply client/player.ts:1]: Does it need to stay or can you delete it? Don't want people to accidently use it for anything.

**Review by RoryT-U — APPROVED** [review]

**Pull request merged**